pywbem/cim_obj.py

```diff
@@ -2511,6 +2511,72 @@ def to_wbem_uri(self, format='standard'):
 
         return _ensure_unicode(''.join(ret))
 
+    @staticmethod
+    def from_instance(class_, instance, namespace=None, host=None,
+                      strict=False):
+        """
+        Given a class and corresponding instance, create and return the
+        CIMInstanceName from the class key properties and instance key
+        property values.
+
+        If a property value does not exist in the instance it gets the
+        value from the class default_value unless strict is set. If strict
+        is True, all key properties must exist in the instance
+
+        Parameters:
+
+            class_ (:class:~pywbem.CIMClass`):
+                CIMClass from which the key properties will be retrieved.
+
+            instance: (:class:~pywbem.CIMInstance`):
+                CIMInstance containing the key property values.
+
+            namespace: (:term:``string`):
+                Namespace to include in created CIMInstanceName or None.
+
+            host:  (:term:``string`):
+                Host name to include in created CIMInstanceName or None.
+
+            strict: (:class:`py:bool`):
+                If True, all key properties are required in instance. Default
+                False
+
+        Returns:
+            :class:`CIMInstanceName` built from the key properties in the
+            class using the key property values in the instance.
+
+        Raises:
+            ValueError: Class and instance names do not match or the
+            strict attribute is True and key property does not exist in
+            the instance.
```

Comment:
Most likely, the first line of this description paragraph needs to be outdented by 2 from the rest of the lines, to make Sphinx generate one paragraph out of this. Look at the generated documentation.